### ISSUE TITLE
auth - Memory Leak

### ISSUE THREAD

**Issue by avickers** [firebase/firebase-js-sdk#1035]
### [REQUIRED] Describe your environment

  * Operating System version: OS X 10.13
  * Browser version: Google Chrome 67 (and Firefox)
  * Firebase SDK version: 5.1.0 & 5.3.0
  * Firebase Product: auth

### [REQUIRED] Describe the problem

onAuthStateChanged slowly consumes more and more memory, some of which is not garbage collected.  Calling the ostensibly returned unsubscribe function does not stop the problem.

Chrome's allocation tools confirm the only allocations are from auth.esm.js

#### Steps to reproduce:

I opened another project that used Firebase and not auth and confirmed the memory footprint was stable.  Added auth and onAuthStateChanged, and the memory leak occurred.  Immediately calling the returned unsubscribe did not stop the leaking.

#### Relevant Code:

```javascript
  let unsub = firebase.auth().onAuthStateChanged(user => {
    if (user) {
    } else {
    }
  })
 unsub()
```

Calling unsub within the callback or after makes no difference.  Something isn't getting disposed of properly.

**Comment by bojeil-google** [comment]
There are a lot of things that go on under the hood for Auth. They are not related to `onAuthStateChanged` listener. Make sure you compare this to not calling the `onAuthStateChanged` listener to make sure this is the real cause.
I will look into replicating this when I get the chance.

**Comment by avickers** [comment]
You are right.  Just calling firebase.auth() kicks off the leak.

**Comment by avickers** [comment]
Resolved by d2f0667fceb4bb2cbc8b08f536bf3785d1de9801
